### DISCUSSION TITLE
Add a configuration setup page to configure the theme settings

### DISCUSSION THREAD

**Discussion by kuige** [squidfunk/mkdocs-material#2805] [Ideas]
MkDocs-material has so many configuration settings, manually editing the yaml config file is not so convenient than a visual setup page, is it possible to have a setup page to configure everything, and the settings are automatically saved to the yaml config file ?

Additionally, with the setup page, one can configure many settings without reading the documents first, which saves time.

**Comment by squidfunk** [comment]
Thanks for suggesting. That could indeed be helpful to wade through the long list of configuration options, but it would be a lot of work to pull off. I can't really say whether this would be worth the effort and I'm afraid I won't have the time to implement this myself because we already have a pretty packed roadmap.

Nonetheless, the documentation should be read as follows:

- __Getting started__ (audience: mainly devs): General setup, minimal configuration, and deployment (= build an MVP)
- __Setup__ (audience: devs + writers): A list of all configuration and customization options to match your brand identity
- __Reference__ (audience: mainly writers): A list of all Markdown constructs supported by Material for MkDocs

The aim of the documentation is to provide a clear structure. However, it can't save you the time of going through the list of options in case you want to customize it (some users don't), because all settings need at least some explanation.
